tests/_temporary_private_credential_api_test.py

```diff
@@ -0,0 +1,55 @@
+"""Tests for the temporarily hosted private credential structure."""
+
+import pytest
+
+from awx_plugins.interfaces._temporary_private_credential_api import (
+    Credential,
+    GenericOptionalPrimitiveType,
+)
+
+
+def test_credential_instantiation() -> None:
+    """Check that credential type can be instantiated."""
+    assert Credential()
+
+
+@pytest.mark.parametrize(
+    ('inputs', 'key', 'expected'),
+    (
+        pytest.param({'foo': 'bar'}, 'foo', 'bar', id='key-present'),
+        pytest.param({'foo1': 'bar1'}, 'baz', None, id='key-missing'),
+    ),
+)
+def test_credential_get_input(
+        inputs: dict[str, GenericOptionalPrimitiveType],
+        key: str,
+        expected: str,
+) -> None:
+    """Check that get_input operates on the dict we provided."""
+    assert Credential(inputs=inputs).get_input(key) is expected
```

Comment:
@chrismeyersfsu oh, you shouldn't have done the same here. The other test has an identity check because both params are bool. In this case, one is a string. The identity check should be used for bool / none but for other types, it should be equality.

This is a bit of a weird case because ideally `None` should be compared using the identity check while `'bar'` needs to use `==`. Still, it shouldn't be a big deal for `None` to also have `==` in tests.